### PR TITLE
Created mnist_classifier.py -  Classifying MNIST dataset using a modified AlexNet

### PR DESCRIPTION
**Pull request by blackeye735**
Python program to train and test a MNIST dataset using a modified AlexNet model.

The program should accept three integers as 
command line input. First two integers define the labels of the two-class classification and third integer represents the number of epochs for training.

### Describe your change:

* [x] Add an algorithm? - Yes [train and test a MNIST dataset using a modified AlexNet model]
* [ ] Fix a bug or typo in an existing algorithm? - [No]
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change? - [No]

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: computer_vision/mnist_classifier.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+import torch.nn as nn
+import torchvision.datasets as dset
+import torchvision.transforms as transforms
+from torch.autograd import Variable
+import torch.nn.functional as F
+import torch.optim as optim
+
+
+class AlexNet(nn.Module):   
+    def __init__(self, num=10):
```
[the remark this answers is not in the document]
Comment (blackeye735) [reply computer_vision/mnist_classifier.py:23]: Function _init_ takes the model AlexNet() and the number of epochs (num=10) and loads the model.
num refers to number of epochs.

File: computer_vision/mnist_classifier.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+            nn.Dropout(),
+            nn.Linear(32*12*12, 2048),
+            nn.ReLU(inplace=True),
+            nn.Dropout(),
+            nn.Linear(2048, 1024),
+            nn.ReLU(inplace=True),
+            nn.Linear(1024, 10)
+        )
+    
+    def forward(self, x):
```
[the remark this answers is not in the document]
Comment (blackeye735) [reply computer_vision/mnist_classifier.py:52]: Forward functon has the forward neural-network comprised for Convolution & MaxPool.
It determines the data-set image size, batch and the volume of neural network to pass through.

'x' is the combination of neural network volume which originated from feature extractor and classifier.

File: computer_vision/mnist_classifier.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        # define forward network 'x' that combines feature extractor and classifier
+        x = self.feature(x)
+        x = x.view(x.size(0), -1)
+        x = self.classifier(x)
+        return x
+
+
+def load_subset(full_train_set, full_test_set, label_one, label_two):
```
[the remark this answers is not in the document]
Comment (blackeye735) [reply computer_vision/mnist_classifier.py:60]: *load_subset* function defines the data that comes from the MNIST dataset.

full_train_set - training MNIST data
full_test_set - test MNIST data
label_one & label_two are the input labels for classification

File: computer_vision/mnist_classifier.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+        if torch.cuda.is_available():
+            data, target = data.cuda(), target.cuda()
+        data, target = Variable(data), Variable(target)
+        optimizer.zero_grad()
+        output = model(data)
+        loss = F.cross_entropy(output, target)
+        loss.backward()
+        optimizer.step()
+
+def test(model,test_loader):
```
[the remark this answers is not in the document]
Comment (blackeye735) [reply computer_vision/mnist_classifier.py]: *test* function calculates the accuracy, loss, output details with the test MNIST data.

model - AlexNet model
test_loader - loading the test data.

File: computer_vision/mnist_classifier.py [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+        if data_lim>0:
+            data_lim-=1
+            if data[1]==label_one or data[1]==label_two:
+                test_set.append(data)
+        else:
+            break
+
+    return train_set, test_set
+
+def train(model,optimizer,train_loader,epoch):
```
[the remark this answers is not in the document]
Comment (blackeye735) [reply computer_vision/mnist_classifier.py]: *train* function calculates the accuracy, loss and output details from train MNIST data.

model - AlexNet model
optimizer - SGD optimizer (Stochastic Gradient Descent)
train_loader - loading the training data
epoch - number of iterations

**Comment by blackeye735** [comment]
The doctests are given. The comments are given and resolved for each function and variables.

**Comment by blackeye735** [comment]
@algorithms-keeper review

**Comment by blackeye735** [comment]
modified the file

**Comment by blackeye735** [comment]
modified the code.

**Comment by cclauss** [comment]
Closing require_type_hints PRs to prepare for Hacktoberfest
